docs/dev/agent_dev_env.md

```diff
@@ -7,82 +7,105 @@ To build the agent on Windows, see [datadog-agent-buildimages](https://github.co
 development files to be available in the dev env. The Agent can embed Python2
 and/or Python3, you will need development files for all versions you want to
 support.
 
-If you're on OSX/macOS, installing Python 2.7 and/or 3.9 with [Homebrew](https://brew.sh)
+If you're on OSX/macOS, installing Python 2.7 and/or 3.11 with [Homebrew](https://brew.sh)
 brings along all the development files needed:
 
 **Please note that not using Python versions explicitly supported, you may have
 problems running the built Agent's Python checks, especially if using a virtualenv.
-At this time, only Python 3.9 is confirmed to work as expected in the development
+At this time, only Python 3.11 is confirmed to work as expected in the development
 environment.**
 ```
 brew install python@2
-brew install python@3.9
+brew install python@3.11
 ```
 
 On Linux, depending on the distribution, you might need to explicitly install
 the development files, for example on Ubuntu:
 ```
 sudo apt-get install python2.7-dev
-sudo apt-get install python3.9-dev
+sudo apt-get install python3.11-dev
 ```
 
-On Windows, install Python 2.7 and/or 3.9 via the [official installer](https://www.python.org/downloads/).
+On Windows, install Python 2.7 and/or 3.11 via the [official installer](https://www.python.org/downloads/).
 
 #### Python Dependencies
 
 ##### Preface
 
-To protect and isolate your system-wide python installation, a python virtual
-environment is _highly_ recommended (though optional). It will help keep a
-self-contained development environment and ensure a clean system Python.
+[Invoke](http://www.pyinvoke.org) is a task runner written in Python that is extensively used in this project to orchestrate builds and test runs. To run the tasks, you need to have it installed on your machine. We offer two different ways to run our invoke tasks.
+
+##### `deva` (recommended)
+
+The `deva` CLI tool is a single binary that can be used to install and manage the development environment for the Agent, built by the Datadog team. It will install all the necessary Python dependencies for you. The development environment will be completely independent of your system Python installation. This tool leverages [PyApp](https://ofek.dev/pyapp/latest/), a wrapper for Python applications that bootstrap themselves at runtime. In our case, we wrap `invoke` itself and include the dependencies needed to work on the Agent.
+
+To install `deva`, you'll need to:
+
+1. Download the binary for your platform from the [releases page](https://github.com/DataDog/datadog-agent-devtools/releases/latest),
+2. Make it executable,
+3. Run the invoke command you need.
+
+The Python environment will automatically be created on the first run. and will be reused for subsequent runs. For example:
+
+```shell
+$ cd datadog-agent
+$ curl -L -o deva https://github.com/DataDog/datadog-agent-devtools/releases/download/deva-v1.0.0/deva-aarch64-unknown-linux-gnu-1.0.0
+$ chmod +x deva
+$ ./deva linter.go
```

Comment:
you simply replaced `inv` by `deva` everywhere. Should it be `./deva`? Should you add a note about adding it in the PATH?

Reply:
added!